package.json

```diff
@@ -81,6 +81,8 @@
     "supertest": "^3.0.0",
     "supports-color": "^5.3.0",
     "terser-webpack-plugin": "^1.1.0",
+    "ts-loader": "^5.3.3",
+    "typescript": "^3.2.4",
```

Comment:
I agree that at least `typescript` should be in `peerDependencies`. It looks like `ts-loader` specifies `"typescript": "*"` in its `peerDependencies`, but that is pretty risky. It also would be a little odd to put `ts-loader` in `peerDependencies` when it's technically optional, but there is no official support for optional `peerDependencies` in npm/yarn.

Reply:
Was talking to @Druotic and he brought up a good point. This is kind of an awkward situation because ideally this would be a `peerDependency`, but some projects are not using TypeScript. This wouldn't technically cause breaking functionality, but it would cause a warning in the console. I'm not sure what the right direction is to move here.

Reply:
Also, there could be projects using `lambda-tools` with TypeScript 2. I think most of our projects have been moved over to TypeScript 3, but there could be a few hanging ones.

Reply:
As @austinkelleher pointed out - by adding typescript to the list of peer dependencies, suddenly all projects consuming this project will need to add typescript to their list of dependencies (i.e. the majority of existing LO services). From the perspective of a consumer project that knows nothing about typescript, that seems odd.

I admit to being less familiar with typescript/ts-loader, so I'm not sure if there's a better workaround for this kind of problem. The first thing that jumps to mind is some kind of plugin system and all typescript specific logic should really live there. Lots of libraries have their core project standalone and lots of different flavoring plugins to support various developer tooling/language choices.

This, of course, is a much bigger effort and will likely get shot down - we might just have to live with the warning. As a middle-ground, perhaps we could add a post-install hook to log a message to give the user more context about the unmet peer dependency warning (that there's no cause for concern and isn't necessary unless using typescript)?

Reply:
I just updated `typescript` and `ts-loader` to be `optionalDependencies`. In the build file ([here](https://github.com/lifeomic/lambda-tools/pull/58/files#diff-78d6e474ae315d0bea76cd46368acfedR67)), I'm now trying to resolve those dependencies if a `tsconfig` file is specified, and exiting if they aren't installed. This is similar to how [react-scripts](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/scripts/utils/verifyTypeScriptSetup.js#L55) handles optionally using TypeScript.